### ISSUE TITLE
[BUG] SAINT's RowColTransformer skips its LayerNorms and dropouts

### ISSUE THREAD

**Issue by ChrisW09** [OpenTabular/DeepTab#416]
**Describe the bug**
`RowColTransformer` (the backbone SAINT uses) registers pre-attention LayerNorms and dropouts but never applies them. The layers are built as `nn.Sequential(LayerNorm, MultiheadAttention, Dropout)` (`deeptab/nn/blocks/transformer.py:377-421`), but the forward pass indexes only the attention module (`transformer.py:431-440`):

```python
x = attn1[1](x, x, x)[0] + x   # skips attn1[0] (LayerNorm) and attn1[2] (Dropout)
...
x = attn2[1](x, x, x)[0] + x
```

So SAINT trains with **no normalization before either attention block and no attention-path dropout**. Verified: after a backward pass, the LayerNorm weights inside the attention Sequentials (`layers.0.0.0.weight`, `layers.0.2.0.weight`, …) have `grad is None` — they're dead parameters sitting in the optimizer.

**To Reproduce**
```python
import numpy as np, pandas as pd
from deeptab.models import SAINTRegressor

X = pd.DataFrame({"a": np.random.randn(80), "b": np.random.randn(80)})
y = np.random.randn(80)
m = SAINTRegressor()
m.fit(X, y, max_epochs=1, accelerator="cpu")
dead = [n for n, p in m._task_model.named_parameters()
        if "norm" in n.lower() and p.grad is None]
print(dead)  # the pre-attention LayerNorms
```

**Expected behavior**
Pre-norm and dropout should be applied around the attention calls (e.g. `x = attn1[2](attn1[1](*(attn1[0](x),)*3)[0]) + x`), matching the SAINT paper's block structure — or the Sequential should be replaced by explicit named modules so the forward can't silently skip components.

**Screenshots**
n/a

**Desktop (please complete the following information):**
 - OS: macOS (Darwin 25.5.0, arm64)
 - Python version: 3.11.15
 - deeptab Version: 2.0.0 (main @ 4e6a359)

**Additional context**
Consequence is silent: training runs, but with missing normalization SAINT is prone to instability and diverges from the published architecture. A generic regression test asserting every registered parameter receives a gradient after one training step would catch this and the MambaTab dead-backbone bug alike.
